### ISSUE TITLE
Render default template.html with Preact instead ejs

### ISSUE THREAD

**Issue by yavorski** [developit/preact-cli#224]
Hi

I am talking about the `template.html` which is located [here](https://github.com/developit/preact-cli/blob/master/src/resources/template.html).
After general discussion in the slack channel, the suggestions are: 

Put in your `template.html` file

```jsx
<!-- in your template.html file -->
<%= require('preact-render-to-string')( require('./template.js').default(options) ) %
```

Then in `template.js`

```jsx
export default props => (
  <html>
    <head>
      <title>{ props.title }</title>
    </head>
    <body>
      <div id="app" />
    </body>
  </html>
)
```

Some addition, still not tested: 

```jsx
export default props => (
  <html>
    <head>
      <title>{ props.title }</title>
    </head>
    <body>
      <div id="app" dangerouslySetInnerHTML={{
        __html: props.ssr(props)
      }} />
    </body>
  </html>
)
```

And some thoughts from @developit

> Not sure about the default, but we could detect `.js` vs `.html` and do the first part for you
> there isn't a `jsx-to-string loader` for webpack that would handle it internally, so we'd be inventing that bit haha
> another option would be to prerender from the root, which might be even more compelling
> no template at all, your app just has control over `<html>` on both client and server

TLDR;

In general I am opening this issue, so we can have something to refer to, and for reference so i can sort/test this things out a lil bit later.
Looking forward for comments and dicussion.
10x!

**Comment by developit** [comment]
Totally. I think we might be able to do this by inspecting the file extension of the template that gets passed to us - if it's `js` or `jsx`, we'd interpret that as a function that returns either VDOM or HTML, and if it returns VDOM we'd run that through `preact-render-to-string`.  Love this idea!

**Comment by lukeed** [comment]
I've arrived to the same practice in my own preact project right now. :raised_hands: Still working out kinks, but I think it's doing nicely!

**Comment by yavorski** [comment]
Hi fellas!
Finally managed to get to this.
As sugessted, I created `src/template.html` with the following contents

```jsx
<!doctype html>
<%= require('preact-render-to-string')(require('./components/Template.jsx').default({ htmlWebpackPlugin, webpack })) %>
```

And `components/Template.jsx` with the following contents: 

```jsx
import { h } from 'preact';

export default (props) => {
  const { options } = props.htmlWebpackPlugin;
  const { files } = props.htmlWebpackPlugin;
  const { chunks } = props.webpack

  const handleChunks = () => {
    for (let chunk of chunks) {
      if (chunk.names.length === 1 && chunk.names[0] === 'polyfills') continue;
      for (let file of chunk.files) {
        if (options.preload && file.match(/\.(js|css)$/)) {
          <link rel='preload' href={files.publicPath + file} as={file.match(/\.css$/) ? 'style' : 'script'} />
        }
        else if (file.match(/manifest\.json$/)) {
          <link rel='manifest' href={ files.publicPath + file } />
        }
      }
    }
  };

  return (
    <html>
      <head>
        <meta charset='utf-8' />
        <title>{options.title}</title>
        <meta name='viewport' content='width=device-width,initial-scale=1' />
        <meta name='mobile-web-app-capable' content='yes' />
        <meta name='apple-mobile-web-app-capable' content='yes' />
        <link rel='manifest' href='/manifest.json' />
        { options.manifest.theme_color && <meta name='theme-color' content={options.manifest.theme_color} /> }
        { handleChunks() }
      </head>
      <body>
        { options.ssr({ url: '/' }) }
        <script defer src={files.chunks['bundle'].entry}></script>
        <script>window.fetch || document.write('<script src={ files.chunks["polyfills"].entry }></script>')</script>
      </body>
    </html>
  );
};
```

Everything is working nicely and i like it very much!
However i would be happy if someone make code review.
I think it would be awesome if we add this to the `wiki` and point it the `readme` `template` section.
